src/PSR7/PathFinder.php

```diff
@@ -76,12 +77,12 @@ public function search() : array
                     continue;
                 }
 
-                // path matched!
-                $paths[] = $opCandidate['addr'];
+                // path matched! use candidatePath as array index to collect every path only once.
+                $paths[$candidatePath] = $opCandidate['addr'];
```

Comment:
shouldn't we just break the cycle if we find the match? because instead we will spend more cycles overwriting this value

Reply:
Good point, yeah I'll update.